python_ta/checkers/condition_logic_checker.py

```diff
@@ -0,0 +1,102 @@
+"""
+Check for redundant If or While conditions in functions based on z3 constraints
+"""
+
+from typing import Any, Union
+
+try:
+    import z3
+
+    from ..cfg.graph import Z3Environment
+
+    z3_dependency_available = True
+except ImportError:
+    z3 = Any
+    Z3Environment = Any
+    z3_dependency_available = False
+
+from astroid import nodes
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+
+class ConditionLogicChecker(BaseChecker):
+    name = "redundant-condition"
+    msgs = {
+        "R9900": (
+            """This condition will always evaluate to True;
+             consider removing redundant check.""",
+            "redundant-condition",
+            "Used when an If or While statement is always True. Requires the z3 configuration option to be True.",
+        ),
+        "R9901": (
+            """This condition will always evaluate to False""",
+            "impossible-condition",
+            "Used when an If or While statement is always False. Requires the z3 configuration option to be True.",
+        ),
+    }
+    options = (
+        (
+            "z3",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y or n>",
+                "help": "Use Z3 to perform logical feasibility analysis in program control flow.",
+            },
+        ),
+    )
+
+    @only_required_for_messages("redundant-condition", "impossible-condition")
+    def visit_if(self, node: nodes.If) -> None:
+        """Visit if statement"""
+        self._check_condition(node)
+
+    @only_required_for_messages("redundant-condition", "impossible-condition")
+    def visit_while(self, node: nodes.While) -> None:
+        """Visit while statement"""
+        self._check_condition(node)
+
+    def _check_condition(self, node: Union[nodes.If, nodes.While]) -> None:
+        """A condition statement is redundant if for every feasible execution path
+        leading to the node, the condition must be True by precedent constraints.
```

Comment:
This explanation is good, but you should make the following two changes:

1. The first line of the docstring should indicate what the function does, not provide a definition. Something like `Check whether `node.test` is redundant or impossible.`
2. You should include a parallel definition for "impossible" condition, since this is also relevant for this method.